File: ts/2025/day06.ts
let p1 = 0, p2 = 0
let lines: string[] = []
for await (const line of console) {
	if (line === "") { continue }
	lines.push(line)
}

function transpose<T>(matrix: T[][]): T[][] {
	return (matrix[0] === undefined) ? [] : matrix[0].map((_, j) => matrix.map(r => r[j]!))
}

let N = lines.map(line => line.split(/\s+/).filter(Boolean))
for (const col of transpose(N)) {
	const ints = col.slice(0, -1).map(Number)
	const op = col.at(-1)
	p1 += op === "+" ?
		ints.reduce((m, v) => m + v, 0) :
		ints.reduce((m, v) => m * v, 1)
}

N = transpose(lines.map(line => line.split("")))
let fs = true, OP = "", R = 0
for (const [i, col] of N.entries()) {
	const num = Number(col.slice(0, -1).join(""))
	const op = col.at(-1)!
	if (fs && op != " ") {
		OP = op
		R = OP === "+" ? 0 : 1
		fs = false
	}
	if (num === 0) {
		fs = true
	} else {
		if (OP === "+") {
			R += num
		} else {
			R *= num
		}
	}
	if (fs || i === N.length - 1) {
		p2 += R
	}
}

console.log("Part 1: %d", p1)
console.log("Part 2: %d", p2)

File: py/2025/day06.py
import sys
import math

p1, p2 = 0, 0
D = sys.stdin.read()
lines = [line for line in D.split("\n") if line]

N = tuple(zip(*(line.split() for line in lines)))
for *nums, op in N:
    ints = [int(n) for n in nums]
    p1 += sum(ints) if op == "+" else math.prod(ints)

N = tuple(zip(*tuple(lines)))
fs, OP, R = True, "", 0
for i, (*nums, op) in enumerate(N):
    if fs and op != " ":
        OP = op
        R = 0 if OP == "+" else 1
        fs = not fs
    if all(d == " " for d in nums):
        fs = True
    else:
        v = int("".join(nums))
        if OP == "+":
            R += v
        else:
            R *= v
    if fs or i == len(N) - 1:
        p2 += R

print(f"Part 1: {p1}")
print(f"Part 2: {p2}")
